src/praisonai/praisonai/cli/features/job_workflow.py

```
@@ -294,31 +294,50 @@ def _exec_python_script(self, script_path: str, step: Dict) -> Dict:
-            exec(code, namespace)
-            return {"ok": True, "output": namespace.get("result", "")}
+            # Parse as expression only (no statements like exec, import, etc.)
+            tree = ast.parse(code.strip(), mode="eval")
+            
+            # Validate all nodes are in allowlist
+            for node in ast.walk(tree):
+                if not isinstance(node, _ALLOWED_NODES):
+                    return {"ok": False, "error": f"Disallowed expression node: {type(node).__name__}. Only simple expressions are allowed."}
+            
+            # Evaluate with empty builtins (no access to getattr, type, etc.)
+            result = eval(compile(tree, "<workflow>", "eval"), {"__builtins__": {}}, namespace)
```

Ruff (0.15.12)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)